test/e2e/hypernode/network_topology_test.go

```diff
@@ -0,0 +1,463 @@
+						Img:         e2eutil.DefaultNginxImage,
+						Req:         e2eutil.CPU2Mem2,
+						Min:         2,
+						Rep:         2,
+						Tolerations: tolerations,
+					},
+				},
+			}
+			topologyJob := e2eutil.CreateJob(ctx, job)
+
+			defer func() {
+				By("Delete job")
+				Expect(ctx.Vcclient.BatchV1alpha1().Jobs(topologyJob.Namespace).Delete(context.TODO(), topologyJob.Name, metav1.DeleteOptions{})).NotTo(HaveOccurred())
+			}()
+
+			By("Wait for job running")
+			Expect(e2eutil.WaitJobReady(ctx, topologyJob)).NotTo(HaveOccurred())
+
+			By("Verify pods are scheduled to tier 2 s5")
+			Expect(e2eutil.VerifyPodScheduling(ctx, topologyJob, []string{"kwok-node-5", "kwok-node-7"})).NotTo(HaveOccurred())
+		})
+
+		It("Case 2.2: pods are pending when tier 2 resources are insufficient", func() {
+			podSpecs := []e2eutil.PodSpec{
+				// make sure tier 1 and tier 2 has insufficient resources, tier 3 has enough resources
+				{Name: "pod-0", Node: "kwok-node-0", Req: e2eutil.CPU4Mem4, Tolerations: tolerations},
+				{Name: "pod-1", Node: "kwok-node-1", Req: e2eutil.CPU4Mem4, Tolerations: tolerations},
+				{Name: "pod-2", Node: "kwok-node-2", Req: e2eutil.CPU4Mem4, Tolerations: tolerations},
+				{Name: "pod-3", Node: "kwok-node-4", Req: e2eutil.CPU4Mem4, Tolerations: tolerations},
+				{Name: "pod-4", Node: "kwok-node-5", Req: e2eutil.CPU1Mem1, Tolerations: tolerations},
+				{Name: "pod-5", Node: "kwok-node-6", Req: e2eutil.CPU1Mem1, Tolerations: tolerations},
+			}
+
+			pods := make([]*v1.Pod, len(podSpecs))
+			for i, podSpec := range podSpecs {
+				pods[i] = e2eutil.CreatePod(ctx, podSpec)
+			}
+
+			defer func() {
+				for _, pod := range pods {
+					Expect(ctx.Kubeclient.CoreV1().Pods(pod.Namespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})).NotTo(HaveOccurred())
+				}
+			}()
+
+			By("Wait for all pods to be ready")
+			for _, pod := range pods {
+				Expect(e2eutil.WaitPodReady(ctx, pod)).NotTo(HaveOccurred())
+			}
+
+			job := &e2eutil.JobSpec{
+				Name: "job-1",
+				NetworkTopology: &batchv1alpha1.NetworkTopologySpec{
+					Mode:               batchv1alpha1.HardNetworkTopologyMode,
+					HighestTierAllowed: ptr.To(2),
+				},
+				Tasks: []e2eutil.TaskSpec{
+					{
+						Name:        "task-1",
+						Img:         e2eutil.DefaultNginxImage,
+						Req:         e2eutil.CPU2Mem2,
+						Min:         2,
+						Rep:         2,
+						Tolerations: tolerations,
+					},
+				},
+			}
+			topologyJob := e2eutil.CreateJob(ctx, job)
+
+			defer func() {
+				By("Delete job")
+				Expect(ctx.Vcclient.BatchV1alpha1().Jobs(topologyJob.Namespace).Delete(context.TODO(), topologyJob.Name, metav1.DeleteOptions{})).NotTo(HaveOccurred())
+			}()
+
+			By("Verify pods are pending")
+			Expect(e2eutil.WaitTaskPhase(ctx, topologyJob, []v1.PodPhase{v1.PodPending}, 2)).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("Soft Mode Tests", func() {
+		It("Case 3: Schedule to single hypernode in soft mode", func() {
```

Comment:
I think there is something wrong with this soft mode case. The reason for scheduling to kwok-node-5 and kwok-node-7 is only because of the influence of binpack plugin. There is already a `network-topology-aware` plugin: https://github.com/volcano-sh/volcano/blob/network-topology/pkg/scheduler/plugins/network-topology-aware/network_topology_aware.go. If the tier1 hypernode has enough resources to schedule a vcjob, then in soft mode, all pods in a vcjob are expected to be under the same hypernode.

Reply:
If you need to test soft mode, you should also open the `network-topology-aware` plugin in https://github.com/volcano-sh/volcano/blob/network-topology/installer/helm/chart/volcano/config/volcano-scheduler-ci.conf